rmw_fastrtps_shared_cpp/src/qos.cpp

```diff
@@ -142,20 +143,38 @@ bool fill_entity_qos_from_profile(
   return true;
 }
 
+template<typename DDSEntityQos>
+bool fill_data_entity_qos_from_profile(
+  const rmw_qos_profile_t & qos_policies,
+  const rosidl_type_hash_t & type_hash,
+  DDSEntityQos & entity_qos)
+{
+  if (!fill_entity_qos_from_profile(qos_policies, entity_qos)) {
+    return false;
+  }
+  auto user_data_str = rmw_dds_common::encode_type_hash_for_user_data_qos(type_hash);
```

Comment:
This needs to be updated with the new API, and also needs to have error checking.